File: app/src/main/java/sk/romanstrazanec/arkanoid/java/gameobjects/Counter.java
package sk.romanstrazanec.arkanoid.java.gameobjects;

import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;

public class Counter extends GameObject {
    private int num;
    private final int textSize;

    public Counter(float x, float y, int textSize) {
        super(x, y + textSize + 2, Color.WHITE);
        this.textSize = textSize;
        num = 0;
    }

    public void raise() {
        num++;
    }

    public void lower() {
        num--;
    }

    @Override
    public void draw(Canvas canvas, Paint paint) {
        super.draw(canvas, paint);
        paint.setTextSize(textSize);
        canvas.drawText(String.valueOf(num), x, y, paint);
    }
}

File: app/src/main/java/sk/romanstrazanec/arkanoid/gameobjects/Counter.kt
package sk.romanstrazanec.arkanoid.gameobjects

import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint

class Counter(x: Float, y: Float, private val textSize: Int) : GameObject(x, y + textSize + 2, Color.WHITE) {
    private var num = 0
    fun raise() {
        num++
    }

    fun lower() {
        num--
    }

    override fun draw(canvas: Canvas?, paint: Paint) {
        super.draw(canvas, paint)
        paint.textSize = textSize.toFloat()
        canvas!!.drawText(num.toString(), x, y, paint)
    }
}